feat: Customer section added

AdrianAlzamoraG opened the pull request:
Features:
- assets folder added, includes css and the images of the corresponding section.
- main.js file updated
- index file updated

AdrianAlzamoraG committed:
feat: added images to our customers

AdrianAlzamoraG committed:
feat: added new images to our customers

AdrianAlzamoraG committed:
feat: Uploading and updating of files making up the customers section

amenes12 reviewed (commented):
LGTM, great job!

dalbitresb12 commented on index.html:
```diff
@@ -59,10 +62,75 @@
     integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p"
     crossorigin="anonymous"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/Swiper/8.1.3/swiper-bundle.min.js" integrity="sha512-dLuSIV553xT28AiFViM1ZQdOlyts81E3xiQvfiltddvQ0RViD99nY8sNlRx5o/9J18BYoRQV+PHo9dGWRv3vjA==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+
```
Can you remove this blank line?

dalbitresb12 commented on public/assets/css/styles.css:
```diff
@@ -0,0 +1,46 @@
+@import url(./main.css);
```
Can you move all the contents of this file back to `main.css` and remove this one? Please also move the `/public/css` folder back to the root and remove the `/public` folder.

dalbitresb12 commented on index.html:
```diff
@@ -14,7 +14,10 @@
+  <link rel="stylesheet" href="https://unpkg.com/swiper@8/swiper-bundle.min.css"/>
   <link href="styles/main.css" rel="stylesheet">
+  <link href="./public/assets/css/styles.css" rel="stylesheet">
+
```
Remove this blank line

dalbitresb12 replied:
You forgot about this.

dalbitresb12 commented on index.html:
```diff
@@ -14,7 +14,10 @@
     integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
+  <link rel="stylesheet" href="https://unpkg.com/swiper@8/swiper-bundle.min.css"/>
   <link href="styles/main.css" rel="stylesheet">
+  <link href="./public/assets/css/styles.css" rel="stylesheet">
```
We should only have the `styles/main.css` file, we don't need this one.

dalbitresb12 commented on index.html:
```diff
@@ -59,10 +62,75 @@
   </nav>
 
+  <!--Our Customers-->
+  <div class="slider-1">
```
Wrap your component in a `<section>` tag with an id.

dalbitresb12 commented on scripts/main.js:
```diff
@@ -1 +1,38 @@
+setTimeout(function () {
+
+    /* Image Slider - Swiper */
+    var imageSlider = new Swiper(".image-slider", {
```
Avoid using `var`. Prefer `const` (or `let` if you need to reassign its value).

```suggestion
    const imageSlider = new Swiper(".image-slider", {
```

dalbitresb12 commented on scripts/main.js:
```diff
@@ -1 +1,38 @@
 console.log("[main.js] Test main.js");
+setTimeout(function () {
```
Waiting for the [`load` event](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) to fire would work better here.

```suggestion
window.addEventListener("load", () => {
```

dalbitresb12 reviewed (changes requested):
Just minor changes.

dalbitresb12 commented on index.html:
```diff
@@ -59,10 +62,75 @@
+                  <img
+                          class="img-fluid"
+                          src="./public/assets/images/customer-logo-1.png"
+                          alt="alternative"
```
Forgot to tell you to set the `alt` attributes properly (they should include a description of the picture).

dalbitresb12 replied:
This applies to all `<img>` tags you add.

AdrianAlzamoraG committed:
feat: Update main.css file
- Moved content from styles.css file to main.css file

AdrianAlzamoraG committed:
feat: Updated file structure

- I have removed the public folder and the css folder with the style files it contained.

AdrianAlzamoraG committed:
feat: Updated the file structure and the client section based on suggestions.

- Removed empty spaces
- Implemented styling in main.css file
- Wrapped the components in with the section tag and an id
- Other issues were remedied

AdrianAlzamoraG committed:
feat: Main.js file update
- Fixed issues

AdrianAlzamoraG committed:
feat: Removed the public folder and moved the assets folder.

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
+@media (min-width: 1200px) {
+
+  /* General Styles */
+  .container {
```
Same thing about not overriding Bootstrap classes.

AdrianAlzamoraG replied:
Reviewed and corrected

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
 }
+
+
+div {
+  display: block;
+}
```
This is by default in all browsers.
```suggestion
```

AdrianAlzamoraG replied:
Reviewed and corrected

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
+/*    Customers     */
+/********************/
+
+.blue {
```
Would be better if you set the name like this:

```suggestion
.text-blue {
```

AdrianAlzamoraG replied:
Reviewed and corrected

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
+.blue {
+  color: #00cec9;
+}
+.img-fluid {
+  max-width: 100%;
+  height: auto;
+  margin-left: auto;
+  display: block;
+}
```
Bootstrap [already has this](https://getbootstrap.com/docs/5.0/content/images/#responsive-images).

```suggestion
```

AdrianAlzamoraG replied:
Reviewed and corrected

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
+.slider-1 {
+  padding-top: 2rem;
+  padding-bottom: 3.75rem;
+  background: linear-gradient(rgb(238, 238, 238), rgb(144, 173, 198));
```
I think this looks better without the gradient, but I don't know about everyone else.

@Future-Leaders-SI653/development what do you guys think?

```suggestion
```

AdrianAlzamoraG replied:
Ready!

dalbitresb12 commented on styles/main.css:
```diff
@@ -2,3 +2,97 @@
+  -webkit-transform: translate3d(0px, 0, 0);
+}
+
+.row {
+  --bs-gutter-x: 1.5rem;
+  --bs-gutter-y: 0;
+  display: flex;
+  flex-wrap: wrap;
+  margin-top: calc(var(--bs-gutter-y) * -1);
+  margin-right: calc(var(--bs-gutter-x)/ -2);
+  margin-left: calc(var(--bs-gutter-x)/ -2);
+}
```
Try not to override Bootstrap classes and variables, just apply the changes for your components. You can also change the `margin` properties to the [summarized syntax](https://developer.mozilla.org/en-US/docs/Web/CSS/margin#syntax)

```suggestion
.some-class-name {
  display: flex;
  flex-wrap: wrap;
  margin: 0 -0.75rem;
}
```

dalbitresb12 commented on index.html:
```diff
@@ -58,11 +61,73 @@
+      <div class="row">
+        <div class="col-lg-12">
+          <h4>
+            Trusted by over <span class="blue">1500</span> customers worldwide
```
If you change the class name, don't forget to change it here as well.

```suggestion
            Trusted by over <span class="text-blue">1500</span> customers worldwide
```

dalbitresb12 reviewed (changes requested):
A little bit more of changes, after this I think it's ready for merging.

AdrianAlzamoraG committed:
feat: updated and corrected styles and html files

dalbitresb12 commented on styles/main.css:
```diff
@@ -1,4 +1,62 @@
-/* Override Bootstrap theme */
-:root {
-  --bs-font-sans-serif: 'Roboto', system-ui, -apple-system, "Segoe UI", "Helvetica Neue", Arial, "Noto Sans", "Liberation Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
```
Add this back please.

dalbitresb12 reviewed (changes requested):
Everything else looks okay, just this.

AdrianAlzamoraG committed:
feat: updated and corrected styles and html files

dalbitresb12 reviewed (approved):
LGTM

dalbitresb12 committed:
Merge branch 'main' into feat/our-customers

dalbitresb12 committed:
chore: code formatting and style

dalbitresb12 committed:
chore: moved imgs to correct folder and fixed refs

dalbitresb12 commented:
Just solved a couple of conflicts and code formatting. Also moved images to the correct folder.
